### ISSUE TITLE
508-defect-3 [SCREENREADER]: Screen reader users should hear an alert through `aria-live` after making a selection

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#22622]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** _Josh_

## User Story or Problem Statement

As a screen reader user, I need to know that there is an alert after specific radio button selections. Since I may be accustomed to navigating forms using `tab` , I may accidentally skip relevant information.

## Details

Upon selecting "no" on the supporting evidence page, an alert will appear notifying the user that they need to submit their service treatment records as soon as possible. 

Although this is currently being announced to screen reader users via `aria-describedby` as accomplished in [ticket 19660](https://github.com/department-of-veterans-affairs/va.gov-team/issues/19660), it may be skipped over by advanced screen reader users who may not wait to hear the full announcement.

## Acceptance Criteria

- [ ] The alert is announced to the screen reader user through `aria-live="polite"`

## Environment

* Operating System: any
* Browser: any
* Screenreading device: any
* Server destination: production

## Steps to Recreate

1. Reach step 3 of 5: Supporting evidence
2. Start your screen reader device of choice
2. Select "no"
3. While the "no" option is being read out, press `tab` again to jump to the continue button
4. Confirm that the alert is not read out

## Solution 

An empty live-region could be added to exist on the page at render. Upon selecting "no" it can be populated with the alert, which would then be announced to the user even after they `tab` off the "no" option. This may not be technically feasible with the form system, but would be a more appropriate use of aria in this scenario. It also better ensures that the user will hear the alert.

<img width="1397" alt="Screen Shot 2021-02-09 at 11 14 03 AM" src="https://user-images.githubusercontent.com/14154792/107393011-46dfaf00-6ac8-11eb-8550-12e36fa727d9.png">

## WCAG or Vendor Guidance (optional)

* [WCAG 2.1 Level A - 1.3.2 Meaningful Sequence](https://www.wuhcag.com/meaningful-sequence/) - Users who rely on assistive technology (such as a screen reader) to interpret content, require content to be presented in a meaningful order. If this is presented out of sequence, users may become disorientated and will not understand the content.
* [Gov.uk's check a service is eligible pattern](https://design-system.service.gov.uk/patterns/check-a-service-is-suitable/) would be a significantly more accessible and usable pattern to address this need. Until more testing is conducted with screen readers, the wizard pattern should be used cautiously.

**Comment by rocio-desantiago** [comment]
URL where this occurs:
http://localhost:3001/disability/file-disability-claim-form-21-526ez/supporting-evidence/service-treatment-records

**Comment by rocio-desantiago** [comment]
Testing with screenreader on 10/3

**Comment by jaredcunha** [comment]
For testing this user, be sure to indicate that you **active duty** and that you select the link to the **BDD** form.

**Comment by jaredcunha** [comment]
@joshkimux Question about the AC. Should the `aria-describedby` attribute be removed? What's happening in VoiceOver when we add `aria-live="polite"` is that the alert content is read twice. But, if we were to remove the `aria-describedby` attribute, and the user misses the announcement when making a selection, the association between the radio option and the alert content is lost.

**Comment by joshkimux** [comment]
@jaredcunha ooph, this is a tricky one. If this information is important, I wouldn't rust `aria-live="polite"` on its own to get the job done (just in case it's accidentally missed the first time). Using `aria-describedby` will keep the association, even if the user returns to this page (and `aria-live="polite"` isn't triggered).

**Comment by jaredcunha** [comment]
@joshkimux [Check this out](https://bugs.chromium.org/p/chromium/issues/detail?id=946653). I guess add a little more weirdness to it, when `aria-describedby` content updates, Chromium treats it like a live region. Might need to dig into this a little more, but i wonder if we might have use both attributes and put with with the double reading? I'll have to test this with other browsers.

**Comment by rocio-desantiago** [comment]
Testing in safari now.

**Comment by joshkimux** [comment]
hey @jaredcunha ! update! I met with @rocio-desantiago who was patient enough to test this out with me on VO using iOS. 

**We confirmed a couple of things together including:**
- Adding an `aria-live="polite"` does in fact read this out earlier _before_ the legend as [I documented in this ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/19660) (and proceeded to forget about, thanks to @rocio-desantiago for reminding me of my own documentation); which is valuable in the use case of advanced screen readers who may not wait long enough to hear the announcement.
- Keeping the `aria-describedby` is also important in the use case that someone returns to this but doesn't hear the announcement anymore as it programmatically links the "no" option to the announcement.

**Logically, then, we thought of using both _but_:**
- Doing so creates the double read out announcement (which is really verbose)
- It also uses `aria` twice which isn't great as the first rule of `aria` is don't use `aria` unless you need to

**Which led us to ask, what if we just changed the design so the alert is a part of the tab order so:**
- We don't need any `aria` (fulfilling the first rule of `aria`)
- All screen reader users regardless of experience will encounter it
- We can reduce the amount of content on the page

**Essentially, instead of this pattern where the full alert's content displays with no interaction**
<img width="536" alt="Screen Shot 2022-10-14 at 4 10 04 PM" src="https://user-images.githubusercontent.com/14154792/195933517-04ebd1fd-e558-4198-98f1-4aff33dfd50c.png">

**What if we were to use the design system's expandable alert component (which is still experimental)?**

<img width="551" alt="Screen Shot 2022-10-14 at 4 13 56 PM" src="https://user-images.githubusercontent.com/14154792/195934335-d9b4b78e-f01b-4945-83f5-0ee139f21812.png">

Would love a designer's take on this too @Evafloyd if you have some time too 🙏

**Comment by joshkimux** [comment]
hey folks, thanks to @rocio-desantiago and @jaredcunha 's feedback, I've updated the original ticket (and will continue to do so for future tickets) which includes a change log.

Let me know if y'all have any more feedback on how I can make these easier to parse through! Appreciate everyone here ❤️
